sqlglot/parser.py

```diff
@@ -3162,6 +3161,9 @@ def _parse_table(
         # Postgres supports a wildcard (table) suffix operator, which is a no-op in this context
         self._match_text_seq("*")
 
+        if self._match(TokenType.PARTITION, advance=False):
+            this.set("partition", self._parse_partition())
+
```

Comment:
I think this messes up the following query (valid in duckdb, postgres):

```sql
with t as (select 1 as c) select partition.c from t partition
```

Reply:
Hmm, `TokenType.PARTITION` is already a valid identifier token. Would it suffice to `parse_partition()` only if the token is followed by `L_PAREN`, otherwise retreat?

Reply:
I don't think so, you can name columns in the alias so it'd be ambiguous:

```
D with t as (select 1 as c) select partition.d from t partition (d);
┌───────┐
│   d   │
│ int32 │
├───────┤
│     1 │
└───────┘
```

I think we need to conditionally do this only for dialects that support this clause.